runtime/hsa-runtime/inc/hsa_ext_amd.h

```diff
@@ -2828,6 +2829,102 @@ typedef enum hsa_amd_queue_priority_s {
 hsa_status_t HSA_API hsa_amd_queue_set_priority(hsa_queue_t* queue,
                                                 hsa_amd_queue_priority_t priority);
 
+/**
+ * @brief Queue creation attributes.
+ */
+typedef enum {
+  /**
+   * The queue packet buffer and the queue struct should be allocated in
+   * the agent's device memory.
+   */
+  HSA_AMD_QUEUE_FLAG_DEVICE_MEM = (1 << 0),
+  /**
+   * Used to indicate if the queue created in device memory should be
+   * cacheable.
+   */
+  HSA_AMD_QUEUE_FLAG_CACHEABLE = (1 << 1),
```

Comment:
We should probably discuss if this is even doable at the moment, to flush L2 is costly and if we are writing a packet (aka 64bytes), we can probably also have false sharing issues/

Reply:
Removed